odxtools/nameditemlist.py

```diff
@@ -40,7 +40,14 @@ def append(self, item: T):
         self._list.append(item)
 
         item_name = self._item_to_name_fn(item)
-        assert item_name.isidentifier()
+
+        if not item_name.isidentifier():
+            warnings.warn(f"For NamedItemList objects to work properly, all "
+                          f"item names must be valid python identiers."
+                          f"(Encountered name {item_name} which is not an "
+                          "identifier!")
+            return
```

Comment:
Remove the return, user can still use `getattr(obj, "44&weirdname")` as a workaround.

Reply:
good point. `obj.get("44&weirdname")` and `obj["44&weirdname"]` should work as well...